pandas/core/frame.py

```diff
@@ -4847,6 +4847,10 @@ def apply(self, func, axis=0, broadcast=False, raw=False, reduce=None,
             while guessing, exceptions raised by func will be ignored). If
             reduce is True a Series will always be returned, and if False a
             DataFrame will always be returned.
+        result_type : str {'infer'}, optional
+            * infer : when axis=1 list-like results will be turned into columns
```

Comment:
What are the valid options here? None and 'infer', or can use provide a type to ensure a specific type is returned?

Also could you clarify: is this the result_type per row, or for the entire thing? I assume per row.